beacon/beacon-error.js

```diff
@@ -0,0 +1,30 @@
+"use strict";
+
+if (!this.document) {
+    importScripts("/resources/testharness.js");
+    importScripts("/common/utils.js");
+    importScripts("beacon-common.js?pipe=sub");
+}
+
+test(function() {
+    // Payload that should cause sendBeacon to return false because it exceeds the maximum payload size.
+    var exceedPayload = Array(maxPayloadSize + 1).fill('z').join("");
+
+    var success = navigator.sendBeacon("http://doesnotmatter", exceedPayload);
+    assert_false(success, "calling 'navigator.sendBeacon()' with payload size exceeding the maximum size must fail");
+}, "Verify calling 'navigator.sendBeacon()' with a large payload returns 'false'.");
+
+test(function() {
+    var invalidUrl = "http://invalid:url";
+    // http://osgvsowi/7877590 - Fetch consistently throws TypeMismatchError instead of TypeError
+    assert_throws("TypeMismatchError", function() { navigator.sendBeacon(invalidUrl, smallPayload); },
```

Comment:
I'm leaning towards `TypeError` because we'll also be telling folks to use Fetch for customized cases (e.g. non-POST method) of beacon..

@toddreifsteck wdyt?